romancal/tweakreg/tweakreg_step.py

```diff
@@ -134,460 +141,331 @@ def process(self, input):
-
-        # create WCSImageCatalog object:
-        refang = image_model.meta.wcsinfo
-        # TODO: create RSTWCSCorrector in tweakwcs
-        im = JWSTWCSCorrector(
-            wcs=image_model.meta.wcs,
-            wcsinfo={
-                "roll_ref": refang["roll_ref"],
-                "v2_ref": refang["v2_ref"],
-                "v3_ref": refang["v3_ref"],
-            },
-            meta={
-                "catalog": catalog,
-                "group_id": image_model.meta.group_id,
-                "name": model_name,
-            },
+    def get_tweakreg_catalog(self, source_detection, image_model, index):
+        """
+        Retrieve the tweakreg catalog from source detection.
+
+        This method checks the source detection metadata for the presence of a
+        tweakreg catalog data or a string with its name. It returns the catalog
+        as a Table object if either is found, or raises an error if neither is available.
+
+        Parameters
+        ----------
+        source_detection : object
+            The source detection metadata containing catalog information.
+        image_model : DataModel
+            The image model associated with the source detection.
+        index : int
+            The index of the image model in the collection.
+
+        Returns
+        -------
+        Table
+            The retrieved tweakreg catalog as a Table object.
+
+        Raises
+        ------
+        AttributeError
+            If the required catalog information is missing from the source detection.
+        """
+        if getattr(source_detection, "tweakreg_catalog", None):
+            tweakreg_catalog = Table(np.asarray(source_detection.tweakreg_catalog))
+            del image_model.meta.source_detection["tweakreg_catalog"]
+            return tweakreg_catalog
+
+        if getattr(source_detection, "tweakreg_catalog_name", None):
+            return self.read_catalog(source_detection.tweakreg_catalog_name)
+
+        raise AttributeError(
+            "Attribute 'meta.source_detection.tweakreg_catalog' is missing. "
+            "Please either run SourceDetectionStep or provide a custom source catalog."
         )
 
-        return im
-
-
-def _common_name(group):
-    file_names = []
-    for im in group:
-        if isinstance(im, rdm.DataModel):
-            file_names.append(os.path.splitext(im.meta.filename)[0].strip("_- "))
-        else:
-            raise TypeError("Input must be a list of datamodels list.")
+    def do_relative_alignment(self, imcats):
+        """
+        Perform relative alignment of images.
+
+        This method performs relative alignment with the specified parameters,
+        including search radius, separation, and fitting geometry.
+
+        Parameters
+        ----------
+        imcats : list
+            A list of image catalogs containing source information for alignment.
+
+        Returns
+        -------
+        None
+        """
+        tweakreg.relative_align(
```

Comment:
Nice! This method helps to hide the mapping between the step spec and stcal. Good idea!